File: crates/daemon/src/run_status.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum RunStatus {
    Queued,
    Running,
    Passed,
    Failed,
    Canceled,
}

impl RunStatus {
    pub fn is_finished(self) -> bool {
        matches!(self, Self::Passed | Self::Failed | Self::Canceled)
    }

    pub fn label(self) -> &'static str {
        match self {
            Self::Queued => "queued",
            Self::Running => "running",
            Self::Passed => "passed",
            Self::Failed => "failed",
            Self::Canceled => "canceled",
        }
    }
}

File: ui/src/lib/run-status.ts
import type { RunStatus } from "./api/types";

export function statusColor(status: RunStatus): string {
  switch (status) {
    case "passed":
      return "success";
    case "failed":
      return "error";
    case "running":
      return "processing";
    case "queued":
      return "warning";
    case "canceled":
      return "default";
  }
}
